src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -56,6 +59,17 @@ public HttpsConnectionMiddleware(ConnectionDelegate next, HttpsConnectionAdapter
                 }
             }
 
+            if (options.HttpProtocols == HttpProtocols.Http1AndHttp2)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
```

Comment:
What about Http1AndHttp2 on Windows 7 or earlier? Should we downgrade those to Http1 as well due to missing ALPN support. This wasn't discussed in the linked issue so I'm not sure if it's an oversight or there's a reason we shouldn't be downgrading.

Reply:
Win7 downgrades by default because of the missing ALPN support, we didn't have to do anything explicit.